lib/rules/no-misleading-character-class.js

```diff
@@ -159,20 +175,23 @@ module.exports = {
             }
 
             const foundKinds = new Set();
+            const offsetForLongUnicode = node.parent.type === "NewExpression" ? 2 : 1;
 
             visitRegExpAST(patternNode, {
                 onCharacterClassEnter(ccNode) {
                     for (const chars of iterateCharacterSequence(ccNode.elements)) {
                         for (const kind of kinds) {
-                            if (hasCharacterSequence[kind](chars)) {
-                                foundKinds.add(kind);
+                            const matchingChars = accumulate(chars, characterSequenceIndexFilters[kind]);
+
+                            if (matchingChars.length) {
+                                foundKinds.add({ chars: matchingChars, kind });
```

Comment:
```suggestion
                                foundKinds.push({ chars: matchingChars, kind });
```

If `foundKinds` is an array.